### ISSUE TITLE
Add testing for Python 3.6

### ISSUE THREAD

**Issue by stevepiercy** [Pylons/pyramid#2748]
Another [alpha of Python 3.6 was released](https://www.python.org/downloads/release/python-360a4/), and the next pre-release of Python 3.6 will be 3.6.0b1, currently scheduled for 2016-09-12.

Should we start testing the 3.6 alpha or betas? If so, what do we need to do in tox?

We'll also need to update docs.

**Comment by tseaver** [comment]
Travis has 3.6 support in its [`nightly` Python](https://docs.travis-ci.com/user/languages/python/#Nightly-build-support).

In `tox.ini`, just add `py36` as an environment, and add [`skip-missing-interpreters = True`](http://tox.readthedocs.io/en/latest/config.html#confval-skip_missing_interpreters=BOOL).

**Comment by mmerickel** [comment]
I would also want `allow_failures` set on the nightly build.

https://docs.travis-ci.com/user/customizing-the-build/#Rows-that-are-Allowed-to-Fail

**Comment by stevepiercy** [comment]
I gave this a few attempts in #2754, and I think I got it right on the third try. Would one of you verify?

Locally I installed `3.6-dev` into `~/.pyenv`, but at first I could not figure out how to get tox to pick it up:

```bash
InvocationError: Failed to get version_info for python3.6: b"pyenv: python3.6: command not found\n\nThe `python3.6' command exists in these Python versions:\n  3.6-dev\n\n"
```

This blog post helped:
https://anil.io/blog/python/pyenv/using-pyenv-to-install-multiple-python-versions-tox/#switching-versions-globally

This command:

```bash
pyenv global 3.5.2 2.7.12 3.4.5 3.3.6 3.6-dev 2.6.9
```

...allowed `tox -e py36` to run and pass all tests locally.
